File: Assets/Scripts/messages/BonusMessage.js
#pragma strict

/**
 * @author: cmdr2 <secondary.cmdr2@gmail.com>
 */

private static var instance : BonusMessage;

/* globals */
private static var MESSAGE_DISTANCE : float = 4.5f; // meters


/* scratchpad */
private var bonusMessageText : TextMesh;


@script RequireComponent(TextMesh)
function Start() {
	bonusMessageText = GetComponent(TextMesh);
	if (bonusMessageText) bonusMessageText.text = '';
}


public static function Instance() : BonusMessage {
	if (instance == null) {
		instance = UnityEngine.Object.FindObjectOfType(BonusMessage);
	}
	return instance;
}


public function Show(bonuses : Array) {
	if (bonusMessageText) {
		bonusMessageText.text = 'Skill Bonus: ' + bonuses[0] + "!\n" +
								'Difficulty Bonus: ' + bonuses[1];
	}
}


public function Hide() {
	if (bonusMessageText) {
		bonusMessageText.text = '';
	}
}

File: Assets/Scripts/messages/BonusMessage.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
/**
 * @author: cmdr2 <secondary.cmdr2@gmail.com>
 */
/* globals */
 // meters
/* scratchpad */
[UnityEngine.RequireComponent(typeof(TextMesh))]
public partial class BonusMessage : MonoBehaviour
{
    private static BonusMessage instance;
    private static float MESSAGE_DISTANCE;
    private TextMesh bonusMessageText;
    public virtual void Start()
    {
        this.bonusMessageText = (TextMesh) this.GetComponent(typeof(TextMesh));
        if (this.bonusMessageText)
        {
            this.bonusMessageText.text = "";
        }
    }

    public static BonusMessage Instance()
    {
        if (BonusMessage.instance == null)
        {
            BonusMessage.instance = (BonusMessage) UnityEngine.Object.FindObjectOfType(typeof(BonusMessage));
        }
        return BonusMessage.instance;
    }

    public virtual void Show(object[] bonuses)
    {
        if (this.bonusMessageText)
        {
            this.bonusMessageText.text = ((("Skill Bonus: " + bonuses[0]) + "!\n") + "Difficulty Bonus: ") + bonuses[1];
        }
    }

    public virtual void Hide()
    {
        if (this.bonusMessageText)
        {
            this.bonusMessageText.text = "";
        }
    }

    static BonusMessage()
    {
        BonusMessage.MESSAGE_DISTANCE = 4.5f;
    }

}